src/main/io/gps.c

```diff
@@ -1100,33 +1099,35 @@ void gpsInitUblox(void)
                 return;
             }
 
-            uint32_t now = millis();
-            // send class message requests every 120ms to test the connection.
-            // cannot assume that the GPS update rate is 100hz, can be 1000hz if the buffer is full
-            static uint32_t delay = 0;
-            if (cmp32(now, delay) > GPS_BAUDRATE_TEST_INTERVAL) {
-                ubloxSendClassMessage(CLASS_MON, MSG_MON_VER, 0);
-                delay = now;
-            }
-
-            // action a response, if we get it
+            // check to see if there has been a response to the version command
             if (gpsData.platformVersion > UBX_VERSION_UNDEF) {
-                // set the gps module to the configured rate               // there was a response at this baud rate!
                 serialPrint(gpsPort, gpsInitData[gpsData.baudrateIndex].ubx);
                 // remember this baud rate in case we re-connect
                 initBaudRateIndex = gpsInitData[gpsData.state_position].baudrateIndex;
-                //  we're done here, let's move the the next state
+                // we're done here, let's move the the next state
                 gpsSetState(GPS_STATE_CHANGE_BAUD);
                 return;
             }
 
-            // no response yet, keep trying...
-            if (cmp32(now, gpsData.state_ts) < GPS_BAUDRATE_CHANGE_DELAY_MS) {
+            // Send the version request five times at GPS_BAUDRATE_TEST_INTERVAL
+            static bool messageSent = false;
+            static uint8_t messageCounter = 0;
+            if (messageCounter < GPS_BAUDRATE_TEST_COUNT) {
+                if (!messageSent) {
+                    ubloxSendClassMessage(CLASS_MON, MSG_MON_VER, 0);
+                    messageSent = true;
+                }
+                if (cmp32(gpsData.now, gpsData.state_ts) > GPS_BAUDRATE_TEST_INTERVAL) {
+                    gpsData.state_ts = gpsData.now;
+                    messageSent = false;
+                    ++ messageCounter;
+                }
                 return;
```

Comment:
This should be (almost) equivalent:
```suggestion
            // Send the version request five times at GPS_BAUDRATE_TEST_INTERVAL
            static uint8_t messageCounter = 0;
            if (messageCounter < GPS_BAUDRATE_TEST_COUNT) {
                if (!messageCounter  || cmp32(gpsData.now, gpsData.state_ts) > GPS_BAUDRATE_TEST_INTERVAL) {
                    ubloxSendClassMessage(CLASS_MON, MSG_MON_VER, 0);
                    gpsData.state_ts = gpsData.now;
                    messageCounter++;
                }
                return;
```

Reply:
thanks